### PR TITLE
BC-4422 - Whitelist frame ancestors

### PR DESCRIPTION
**Pull request by uidp**
# Description
Enhance CSP-configuration to prevent ClickJacking attacks.

## Links to Tickets or other pull requests
Ticket [BC-4422](https://ticketsystem.dbildungscloud.de/browse/BC-4422)
https://github.com/hpi-schul-cloud/schulcloud-client/pull/3231

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by Loki-Afro — APPROVED** [review]

File: ansible/group_vars/develop/cfg.yml [hunk 1]
```diff
@@ -8,4 +8,5 @@ ES_DOMAIN: "https://repository.sc-dev-edusharing.dbildungscloud.dev"
 
 H5P_FRAME_SRC_URLS: "https://rendering.edu-2.services.sc-dev-edusharing.dbildungscloud.dev/ https://rendering.services.sc-dev-edusharing.dbildungscloud.dev/"
 
+BETTERMARKS_CSP_URL: "https://apps-ci00.bettermarks.com"
```
Comment (virgilchiriac) [inline ansible/group_vars/develop/cfg.yml:11]: where is this used?

Comment (virgilchiriac) [inline ansible/group_vars/develop/cfg.yml:11]: should we not define this also in group_vars/all/config?
these locations should be use for overwrite a previously defined value

Comment (Loki-Afro) [reply ansible/group_vars/develop/cfg.yml:11]: no will probably change in the near future when there is that bettermarks proxy in place

Comment (hoeppner-dataport) [reply ansible/group_vars/develop/cfg.yml:11]: It is used in schulcloud_client (ansible/roles/schulcloud-client-core/templates/client-configmap-files.yml.j2) in the CSP-configuration to only allow iframes embedding the schulcloud for bettermarks.

**Review by virgilchiriac — APPROVED** [review]

**Pull request merged**